main.py

```diff
@@ -0,0 +1,258 @@
+"""
+This main python file contains the backend logic of the website, including
+the reception of Cloud Pub/Sub messages and communication with GCS and datastore.
+"""
+
+# Import all necessary libraries.
+import webapp2
```

Comment:
import should be in two sections: standard library and third-party. In this case, your imports should be:

```
import collections
import logging
import os
import urllib

import cloudstorage
from google.appengine.api import images
from google.appengine.ext import ndb
from google.appengine.ext import ndb
import googleapiclient.discovery
```